File: 899-binary-gap/binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        first_one = False
        pos = 0
        cur = 0
        ans = 0
        while n:
            if (n & 1) == 1:
                if first_one:
                    ans = max(ans, cur - pos)

                first_one = True
                pos = cur
                
            # update bit position
            cur += 1
            # update n
            n >>= 1
        
        return ans

File: 899-binary-gap/binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        int last = -1;
        int pos = 0;
        int ans = 0;
        while (n) {
            int curr = n & 1;

            if (curr == 1) {
                if (last != -1) {
                    // update ans
                    ans = max(ans, pos - last);
                }
                last = pos;
            }
            pos++;
            n >>= 1;
        }
        
        return ans;
    }
};